File: java/MostFrequentItemInArray.java
import java.util.Map;
import java.util.HashMap;

public class MostFrequentItemInArray{

    public static int mostFrequent(int[] items){
        int max_count = 1; int max_item = -1;
        Map<Integer, Integer> count = new HashMap<>();

        for(int i : items){
          if(count.containsKey(i))
            count.put(i, count.get(i) + 1);
          else
            count.put(i, 1);

          if(count.get(i) > max_count){
            max_count = count.get(i);
            max_item = i;
          }

        }

        return max_item;
    }

    public static void main(String args[]){
      int[] items = {1, 2, 3, 5, 4, 1, 2, 4};
      System.out.println(mostFrequent(items));
    }

}

File: scala/MostFrequentItemInArray.scala
import scala.collection.mutable.HashMap

class MostFrequentItemInArray{

	def findMostFrequentItemInArray(inputArray: Array[Int]) = {
		var max_count = 1; var max_item = -1
		val dataSet = HashMap.empty[Int, Int]
		
		for(i <- inputArray){
			
			if(dataSet.contains(i))
				dataSet(i) += 1
			else
				dataSet(i) = 1
			
			
			if(dataSet(i) > max_count){
				max_count = dataSet(i)
				max_item = i
			}
			
		}
		max_item
	}

}


object MostFrequentItemInArray extends App{
	val mfi = new MostFrequentItemInArray
	val inputData = Array(1, 2, 3, 5, 4, 2, 1, 2)
	println(mfi.findMostFrequentItemInArray(inputData))
}